pinot-common/src/main/java/com/linkedin/pinot/common/utils/FileUploadUtils.java

```diff
@@ -152,6 +205,30 @@ public static int sendSegmentUri(final String host, final String port, final Str
   }
 
   public static int sendSegmentJson(final String host, final String port, final JSONObject segmentJson) {
+    return sendSegmentJson(host, port, segmentJson, MAX_RETRIES, SLEEP_BETWEEN_RETRIES_IN_SECONDS);
+  }
+
+  public static int sendSegmentJson(final String host, final String port, final JSONObject segmentJson,
+      final int maxRetries, final int sleepTimeSec) {
+    for (int numRetries = 0; ; numRetries++) {
+      try {
+        return sendSegmentJsonImpl(host, port, segmentJson);
+      } catch (Exception e) {
+        if (numRetries >= MAX_RETRIES) {
```

Comment:
This one is not readability issue. If user pass in the maxRetries, we should use that value. If not, then use default value.